deepmd/infer/deep_pot.py

```diff
@@ -205,6 +225,265 @@ def _get_natoms_and_nframes(self, coords: np.ndarray, atom_types: List[int]) ->
+            The map is used to transform the atom indexes after padding to the original indexes.
+        """
+        natoms = natoms_vec.astype(np.int32)[0]
+        nloc_padding = graph_para[3][0]
+        type_count = natoms_vec.astype(np.int32)[2:]
+        type_count_padding = graph_para[3][2:]
+        nall_new = int(idx_mapping_i[0])
+        coords_padding = init_input[0]
+        type_padding = init_input[1]
+        map_padding = init_input[2]
+        nlist_padding = init_input[3]
+        neigh_len = self.ASCEND_NEI_LEN
+        feed_dict_padding = {}
+
+        # generate offset mapping
+        offset_mapping = np.array([0 for ii in range(nall_new)])
+        offset_padding = 0
+        offset = 0
+        assert nloc_padding >= natoms
+        for type_i in range(len(type_count)):
+            offset_mapping[offset : offset + type_count[type_i]] = np.arange(type_count[type_i]) + offset_padding
+            offset += type_count[type_i]
+            offset_padding += type_count_padding[type_i]
+        assert offset == natoms
+        offset_mapping[offset : offset + nall_new - natoms] = np.arange(nall_new - natoms) + offset_padding
+        offset_mapping = np.append(offset_mapping, -1)
+
+        # one frame padding begin
+        coords_padding[3 * offset_mapping[:-1]] = coords_i[np.arange(nall_new) * 3]
+        coords_padding[3 * offset_mapping[:-1] + 1] = coords_i[np.arange(nall_new) * 3 + 1]
+        coords_padding[3 * offset_mapping[:-1] + 2] = coords_i[np.arange(nall_new) * 3 + 2]
+        type_padding[offset_mapping[:-1]] = ntypes_i[np.arange(nall_new)]
+        map_padding[offset_mapping[:nall_new]] = offset_mapping[idx_mapping_i[1:nall_new+1].astype(int)]
+        nlist_padding[nloc_padding + offset_mapping[:natoms] + 1] = nlist_i[natoms + np.arange(natoms) + 1]
+        for ii in range(natoms):
+            nlist_padding[2 * nloc_padding + offset_mapping[ii] * neigh_len + 1: 2 * nloc_padding + (
+                            offset_mapping[ii] + 1) * neigh_len + 1] = offset_mapping[nlist_i[
+                            2 * natoms + ii * neigh_len + 1 : 2 * natoms + (ii + 1) * neigh_len + 1]]
+        mesh = np.append(nlist_padding, map_padding)
+        feed_dict_padding[self.t_coord] = np.reshape(coords_padding, [-1])
+        feed_dict_padding[self.t_type] = np.reshape(type_padding, [-1])
+        feed_dict_padding[self.t_mesh] = np.reshape(mesh, [-1])
+        feed_dict_padding[self.t_box] = np.reshape(box_i, [-1])
+        return feed_dict_padding, offset_mapping
+
+
+    def _run_with_padding(
+        self,
+        t_out: tf.Tensor,
+        feed_dict: dict,
+        nframes: int,
+        atomic: bool
+        ) -> List:
+        """Evaluate the energy, force and virial, Ascend platform needs padding inputs.
+
+        Parameters
+        ----------
+        feed_dict : dict of tensor
+            Session original feed_dict includes coord, type, box, mesh, natoms.
+        t_out : list of tensor
```

Comment:
The type and the order are different from those in line 408. Please check which is right.

Reply:
Thanks, we have fixed it.